src/controls/changeWidth.ts

```diff
@@ -40,9 +40,8 @@ export const changeObjectWidth: TransformActionHandler = (
         target.strokeWidth / (target.strokeUniform ? target.scaleX : 1),
       multiplier = isTransformCentered(transform) ? 2 : 1,
       oldWidth = target.width,
-      newWidth = Math.ceil(
-        Math.abs((localPoint.x * multiplier) / target.scaleX) - strokePadding,
-      );
+      newWidth =
+        Math.abs((localPoint.x * multiplier) / target.scaleX) - strokePadding;
```

Comment:
Since this was avoiding newWidth to go under 1, could you change the line below to 
```target.set('width', Math.max(newWidth, 1));``` ?

Reply:
Also please check if there is a test for changeWidth and extend it to cover this change.

Reply:
@asturur Sure I can make that change but I just want to make sure it's necessary as versions before v6 didn't have that restriction.
Also, I'm not familiar with how the tests are set up but there doesn't appear to be any existing test that run against the changeWidth function. I did try making changes locally to the controls_handlers test script but it's unclear to me how I'm supposed to reference the changeWidth function there.

Reply:
if the previous code was using Math.ceil, it means that any float that was smaller than 1 was rounded to 1.
As a consequence newWidth was never smaller than 1.
Now your changes remove that to obtain smoothness at higher zoom levels but that 1 barrier should stay, and the line below is a perfect place to reimplement it.

A test i can add when the code changes are done

Reply:
I've just committed that change.